Updates for ITKv5.2 code cleanups ... ROUND1

hjmjohnson opened the pull request.

hjmjohnson committed:
COMP: Enable building without python.

hjmjohnson committed:
STYLE: Prefer = default to explicitly trivial implementations

This check replaces default bodies of special member functions with
= default;. The explicitly defaulted function declarations enable more
opportunities in optimization, because the compiler might treat
explicitly defaulted functions as trivial.

Additionally, the C++11 use of = default more clearly expreses the
intent for the special member functions.

hjmjohnson committed:
DOC: Update copyright assignment to NumFOCUS

The mission of NumFOCUS is to promote open
practices in research, data, and scientific
computing.

https://numfocus.org

hjmjohnson committed:
ENH: Add .gitattributes to allow running ITK clang-formatting scripts

```
git filter-branch -f \
--tree-filter "~/ITK/Utilities/Maintenance/clang-format.bash --clang-format ~/Dashboard/src/ITK-clang11/clang-format-Linux --tracked" \
    master..
```

romangrothausmann commented:
I wonder if `CartesianToPolarTransform` and `PolarToCartesianTransform` (basically this module) deserve to be included in main ITK, as they represent the pendant to `AzimuthElevationToCartesianTransform` (i.e. cylindrical vs. spherical transform).

hjmjohnson committed:
ENH: Prefer https vs http for ctest submission.

hjmjohnson committed:
ENH: Add azure CI pipeline.

hjmjohnson committed:
COMP: Azure is deprecating macOS-10.13 support

https://devblogs.microsoft.com/devops/removing-older-images-in-azure-pipelines-hosted-pools/

Use full paths simplifies debugging

thewtex commented on test/CMakeLists.txt:
```diff
@@ -10,7 +10,9 @@ itk_add_test(NAME itkPolarTransformTest
-  EXPRESSION "instance = itk.PolarToCartesianTransform.New()")
-itk_python_expression_add_test(NAME itkCartesianToPolarTransformPythonTest
-  EXPRESSION "instance = itk.CartesianToPolarTransform.New()")
+if(ITK_WRAP_PYTHON)
```
ITK 5 best practice is to put Python tests in `wrapping/test/CMakeLists.txt`.

thewtex reviewed (approved):
Thanks, Hans!
